pkg/driver/nydus/convert.go

```diff
@@ -88,6 +89,17 @@ func convertToNydusLayer(opt nydusify.PackOption, backend backend.Backend) conve
 		if err != nil {
 			return nil, errors.Wrapf(err, "get blob info %s", blobDigest)
 		}
+		if info.Labels == nil {
+			info.Labels = map[string]string{}
+		}
+		// Write a diff id label of layer in content store for simplifying
+		// diff id calculation to speed up the conversion.
+		// See: https://github.com/containerd/containerd/blob/e4fefea5544d259177abb85b64e428702ac49c97/images/diffid.go#L49
+		info.Labels[labels.LabelUncompressed] = blobDigest.String()
+		_, err = cs.Update(ctx, info)
+		if err != nil {
+			return nil, errors.Wrap(err, "update layer label")
```

Comment:
Looks like it can just fall back to the case without diff id If `cs.Update` fails, do we really need to return error?

Reply:
In general update the info in the content store will success, except for something like grpc disconnected. :)